foreshadow/transformers/internals/uncommonremover.py

```diff
@@ -0,0 +1,46 @@
+import numpy as np
+from sklearn.utils import check_array
+from sklearn.utils.validation import check_is_fitted
+from sklearn.base import TransformerMixin, BaseEstimator
+
+
+class UncommonRemover(BaseEstimator, TransformerMixin):
+    """Merges uncommon values in a categorical column to an other value
+
+        Note: Unseen values from fitting will alse be merged.
+
+        Args:
+            threshold (float): data that is less frequant than this percentage will 
+                be merged into a singular unique value
+            replacement (Optional): value with which to replace uncommon values
+    """
+
+    def __init__(self, threshold=0.01, replacement=None):
+        self.threshold = threshold
+        self.replacement = replacement
+
+    def fit(self, X, y=None):
+        """
+        Finds the uncommon values and sets the replacement value
+        """
+        X = check_array(X, dtype=None)
+        self.values_, counts = np.unique(X, return_counts=True)
+        self.merge_values_ = self.values_[np.where(counts <= self.threshold * X.size)]
+        if self.replacement is None:
```

Comment:
I feel the logic to set the default value for replacement here could be problematic. If it is a string column, we could end up with a very long name at the end. If it is numeric, we won't have a way to tell a category is the result of merging. I suggest we simply set the default value to 'DummyEncoder_Other'.

Reply:
I wanted to leave as much information in the data as possible but if you think it is best to go that route, I can do that! I'll make the change.